smartsim/_core/mli/infrastructure/control/requestdispatcher.py

```diff
@@ -0,0 +1,525 @@
+        self._mem_pool = mem_pool
+        """Memory pool used to share batched input tensors with the Worker Managers"""
+        self._perf_timer = PerfTimer(prefix="r_", debug=True, timing_on=True)
+        """Performance timer"""
+
+    def _validate_request(self, request: InferenceRequest) -> bool:
+        """Ensure the request can be processed.
+        :param request: The request to validate
+        :return: True if the request is valid, False otherwise"""
+        if not self._feature_store:
+            if request.model_key:
+                logger.error("Unable to load model by key without feature store")
+                return False
+
+            if request.input_keys:
+                logger.error("Unable to load inputs by key without feature store")
+                return False
+
+            if request.output_keys:
+                logger.error("Unable to persist outputs by key without feature store")
+                return False
+
+        if not request.model_key and not request.raw_model:
+            logger.error("Unable to continue without model bytes or feature store key")
+            return False
+
+        if not request.input_keys and not request.raw_inputs:
+            logger.error("Unable to continue without input bytes or feature store keys")
+            return False
+
+        if request.callback is None:
+            logger.error("No callback channel provided in request")
+            return False
+
+        return True
+
+    def _on_start(self) -> None:
+        self._queue_swap_lock = RLock()
+
+    def _on_iteration(self) -> None:
+
+        if self._incoming_channel is None:
+            raise SmartSimError("No incoming channel for dispatcher")
+
+        try:
+            bytes_list: t.List[bytes] = self._incoming_channel.recv()
+        except Exception:
+            self._perf_timer.start_timings()
+        else:
+            if not bytes_list:
+                exception_handler(
+                    ValueError("No request data found"),
+                    None,
+                    "No request data found.",
+                )
+
+            request_bytes = bytes_list[0]
+            tensor_bytes_list = bytes_list[1:]
+            self._perf_timer.start_timings()
+
+            request = deserialize_message(request_bytes, self._comm_channel_type)
+            if request.input_meta and tensor_bytes_list:
+                request.raw_inputs = tensor_bytes_list
+
+            self._perf_timer.measure_time("deserialize_message")
+            if not self._validate_request(request):
+                return
+
+            self._perf_timer.measure_time("validate_request")
+            self.dispatch(request)
+
+            self._perf_timer.measure_time("dispatch")
+        finally:
+            self.flush_requests()
+            # TODO: implement this
+            # self.remove_queues()
+
+            self._perf_timer.end_timings()
+
+        if self._perf_timer.max_length == 801:
```

Comment:
Ha! This is purely debug code, because I know that in our tests we are going to run 801 requests. It will be removed (or replaced with a better mechanism)